### ISSUE TITLE
Move to agent-based approach

### ISSUE THREAD

**Issue by cvanelteren** [cvanelteren/PlexSim#3]
Currently the adjacency matrix is the agent stack. The bonabeau model has agent moving around on a finite space. In the current implementation the whole grid is assumed to be an agent. A different update scheme needs to be implemented for higher efficiency.

Suggestions

#  Keep track of agent, and update only these.
- pros:
   - more flexibility in agent dynamics
   - higher efficiency in sparse systems
- cons:
   - basically redesign from the ground up -> maybe better for different language too due to jump from memoryviews to other data structures, prolly structs. 

# Keep as is and deal with inefficiency
- pros: no rewrite
- cons: sparse agents yield to high inefficiency

**Comment by cvanelteren** [comment]
This feature is possible under the current v3
